### ISSUE TITLE
color image support

### ISSUE THREAD

**Issue by ckldan520** [Ryanshuai/BM3D_py#1]
Thanks for your good work. Will you consider supporting color image denoising in the future?

**Comment by Ryanshuai** [comment]
I definitely will support color images. I am really busy now. Can you wait for about 1 or 2 months?

**Comment by ckldan520** [comment]
Thank you very much！

**Comment by 442413768** [comment]
Thanks for your work! Can you write more comments to your code ?

**Comment by ckldan520** [comment]
> > Thanks for your good work. Will you consider supporting color image denoising in the future?
> 
> I definitely will support color images. I am really busy now. Can you wait for about 1 or 2 months?

**Comment by ckldan520** [comment]
Why does the line 25 of bm3d_1st_step.py require that the height and width of img_noisy be equal? This will require the height and width of the original input image to be equal. For example, if I input a 256 * 320 image, there will be some bugs in the program.

**Comment by Ryanshuai** [comment]
@ckldan520 
Yeah, you are right. This is a problem. Cause We only need square pictures for experiments. 
You can either wait for me to fix this or fix it by yourself and pull request.
I think I will fix this in one month.

**Comment by Ryanshuai** [comment]
@ckldan520 I fixed the bug that the image height must be equal to the width, images with different widths and heights work fine now.

**Comment by ckldan520** [comment]
> @ckldan520 I fixed the bug that the image height must be equal to the width, images with different widths and heights work fine now.

Thank you very much, it works fine now

**Comment by TongWenQian** [comment]
Thank you for your excellent work. Will you continue to change BM3D to support color images?

**Comment by Ryanshuai** [comment]
> Thank you for your excellent work. Will you continue to change BM3D to support color images?

I don't think I have time to read the paper. I just read some other codes of BM3D, and I believe it only involves two color transforms. But I must verify the code before releasing it. The verification workload is huge. And I have always been trying to refactor this code, and it also takes time.

**Comment by Ryanshuai** [comment]
> Thank you for your excellent work. Will you continue to change BM3D to support color images?

https://webpages.tuni.fi/foi/GCF-BM3D/ 
This is a python implementation that encapsulates CPP. The color transform is not encapsulated in CPP.
You can refer to this one.

**Comment by TongWenQian** [comment]
> > Thank you for your excellent work. Will you continue to change BM3D to support color images?
> 
> https://webpages.tuni.fi/foi/GCF-BM3D/ This is a python implementation that encapsulates CPP. The color transform is not encapsulated in CPP. You can refer to this one.

Thank you very much.
